crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -64,6 +64,24 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/controllers", "Orchestrate the upgrade of controller nodes"
```

Comment:
No, it's not possible for user to choose controller to upgrade. The whole thing about upgrading controller nodes is pretty complicated to orchestrate, as you can see at https://etherpad.nue.suse.com/p/cloud-upgrade-6-to-7 , step 12 (starting at line 154). 
Very shortly:
-  the controllers need to be upgraded one-by-one, not in parallel (that's why the /controller action would take quite a lot time to complete if we'd want from it upgrade of everything)
- the order in which the nodes are going to be upgraded depends on their configuration, on the services which are running there. We have to determine this programmaticaly, not letting user to choose.